website/content/en/docs/best-practices/multi-tenancy.md

```diff
@@ -0,0 +1,131 @@
+---
+title: "Multi-Tenancy Best Practices"
+linkTitle: "Multi-Tenancy"
+weight: 1
+description: This guide describes the best practices concepts to write Operators for Multi-Tenancy solutions.
+---
+
+## Operator Behavioral Guidelines
+
+### NetworkPolicy 
+
+If your Operator creates or manages [NetworkPolicy][k8s-network-policy] configurations ensure that your solution:
+
+* applies fine-grained network policies to the extent that is required for your managed application to function properly
+* applies fine-grained network policies to enable your managed application internal components to communicate among each other
+* allows users to disable network policy management (your operator doesn't deploy [NetworkPolicy][k8s-network-policy] instances at all)
+* does not create `allow traffic from everywhere in the cluster` type policies
+
+Usage of [NetworkPolicy][k8s-network-policy] is popular in multi-tenant cluster to provide an extra layer
+of segregation among tenants within SDN solutions. Users typically customize this extensively with goal to disallow network traffic among unrelated
+tenants. Operators that deploy a `accept all traffic from anywhere in the cluster` style policy, are creating an obstacle in
+pursuing this goal, especially if it cannot be disabled. In security-conscious environments policies like these are not allowed
+in production. In such cases your operator should minimally **have the option to prevent [NetworkPolicy][k8s-network-policy] objects from being
+created** and leave these to the user. In more advanced cases, your operator should create [NetworkPolicy][k8s-network-policy] configurations that
+follow the least-privilege principle as well implement deny-all-allow-some rules.
+
+### Managing Resources
+
+#### Resource Requests
+
+Create resource requests for CPU and memory:
+
+* for your Operator's manager `Deployment` (See `config/manager/manager.yaml`)
+* for any `Pods` and `Deployments` which are created by your Operator
+
+```yml
+  ...
+  # TODO(user): Configure the resources accordingly based on the project requirements.
+  # More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
+  resources:
+    requests:
+      cpu: 10m
+      memory: 64Mi
+    ... 
+```
+
+In order to prevent tenants from consuming all the resources on a cluster or blocking other workloads 
+from being scheduled, many production clusters will define [ResourceQuota][k8s-resources-quotas] configuration.
+[ResourceQuota][k8s-resources-quotas] configuration in a namespace will cause all pods without 
+[Requests and limits][k8s-requests-limits] for `CPU` and `Memory` to be rejected.
```

Comment:
I supplemented the info. Note that how the CPU limits works are defined as well in this document.
Also, we are adding it as Optional in the general recommendations in order we try to reach a common sense.
Also, we highlight a lot here that usually cluster admins does configurations that will limit the max consumption for all pods in an ns which according to k8s docs is usually done for both (CPU and Memory). 

However, I'd prefer here stick with the `ideally` proposed by the original author of this guidelines, Daniel Messer, since it also matches with other sources : 
- https://cloud.google.com/blog/products/containers-kubernetes/kubernetes-best-practices-resource-requests-and-limits
- https://geekflare.com/kubernetes-best-practices/
- https://cheatsheetseries.owasp.org/cheatsheets/Docker_Security_Cheat_Sheet.html#rule-7-limit-resources-memory-cp[%E2%80%A6]le-descriptors-processes-restarts
- https://stackoverflow.com/questions/48907371/what-purpose-do-cpu-limits-have-in-kubernetes-resp-docker
- https://blog.kubecost.com/blog/requests-and-limits/

IMO the main reason to limit is not to allow 1 pod to consume all and in this way affects the others which also can happen for both CPU and memory. Also, see from k8s docs:
- If you specify a CPU limit but do not specify a CPU request: https://kubernetes.io/docs/tasks/configure-pod-container/assign-cpu-resource/#if-you-do-not-specify-a-cpu-limit
- Motivation for CPU requests and limits: https://kubernetes.io/docs/tasks/configure-pod-container/assign-cpu-resource/#motivation-for-cpu-requests-and-limits

**By configuring the CPU requests **and** limits of the Containers that run in your cluster, you can make efficient use of the CPU resources available on your cluster Nodes.** 

So, seems that the above references does **not** go along with your pov that would not be a good practice to limit CPU but is a good practice to limit the memory.

Reply:
I think we could sort it down. So, closing.